src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -289,6 +297,102 @@ def initialize_newborns(self, pop_data: SimulantData) -> None:
+            pop_data.user_data["current_population_index"],
+            query="alive == 'alive' and in_united_states == True and tracked == True",
+        )
+        new_simulants = self.initialize_simulant_link_columns(
+            new_simulants, existing_simulants
+        )
+
+        # We need to do this each time we initialize more simulants, otherwise the
+        # addition of NaNs in household_id before it was initialized makes
+        # the column a float.
+        new_simulants["household_id"] = new_simulants["household_id"].astype(int)
+
+        self.population_view.update(new_simulants[self.columns_created])
+
+    def initialize_non_reference_person_immigrants(self, pop_data: SimulantData) -> None:
+        """
+        Initializes simulants who have newly immigrated into
+        the US into existing, non-GQ households.
+
+        Parameters
+        ----------
+        pop_data
+            The SimulantData on the simulant creation call, which must supply the ACS
+            rows the simulants should be based on.
+        """
+        new_simulants = self.initialize_new_simulants_from_acs(
+            pop_data.user_data["acs_persons"], pop_data
+        ).set_index(pop_data.index)
+        new_simulants["age"] = self.perturb_individual_age(new_simulants)
+
+        existing_simulants = self.population_view.get(
+            pop_data.user_data["current_population_index"],
+            query="alive == 'alive' and in_united_states == True and tracked == True",
+        )
+        existing_household_ids = existing_simulants[
+            ~existing_simulants["household_id"].isin(data_values.GQ_HOUSING_TYPE_MAP)
+        ]["household_id"].unique()
+
+        # TODO: For now, we do a simple random sample of households.
+        # As part of future PUMA perturbation work, we plan to actually use
+        # the PUMA value in the ACS row in choosing the household they should join.
+        # NOTE: This does not do anything to state and PUMA, leaving them whatever they
+        # were for the ACS individual this immigrant is based on!
+        # This is obviously nonsensical, but mirrors the current behavior of domestic
+        # migration, where state and PUMA do not update on move.
+        new_simulants["household_id"] = vectorized_choice(
+            existing_household_ids,
+            n_to_choose=len(new_simulants),
+            randomness_stream=self.randomness,
+            additional_key="household_id",
+        ).astype(existing_simulants["household_id"].dtype)
+
+        # We avoid non-reference-person immigrants having these relationships, because
+        # otherwise they may cause there to be an impossible number of said relationships
+        # within a household, i.e. >2 parents or >1 spouse/partner
+        new_simulants["relation_to_household_head"] = (
+            new_simulants["relation_to_household_head"]
+            .replace(
+                {
+                    "Parent": "Other relative",
```

Comment:
What about father, mother, step-father, etc.? There are more relationships that imply uniqueness than just these. Or We could add a third grandmother, which is also impossible.

Reply:
Mother and father are both "parent", which is already listed here. I think I should add parent-in-law to this list, and to the documentation.

There is no step-father relationship; that would be "other relative" already. The same is true of grandparent. The full list of relationships is here: https://github.com/ihmeuw/vivarium_census_prl_synth_pop/blob/3f7cf37eef13608d6dd9e246bcce6e4a201cb43f/src/vivarium_census_prl_synth_pop/constants/metadata.py#L150-L170